### ISSUE TITLE
v2.0.9 selectOption() throws InvalidArgumentException: Unreachable field ""

### ISSUE THREAD

**Issue by daison12006013** [Codeception/Codeception#2062]
So I am using a chrome extension to get the exact xpath, I was able to use ``grabTextFrom()`` function to show the lists of options in the select tag. When trying to use ``grabValueFrom()`` it returns a ``NULL``.

Here's my codeception code
```
$el = "//div[@class='formBlock']/div[@class='row form-group'][2]/div[@class='col-sm-4'][1]/div[@class='form-group']/select[@class='citySelection form-control input-sm']";
//dd($this->I->grabTextFrom($el));
$city_name = $this->cities[array_rand($this->cities)];
$city = City::where('name', '=', $city_name)->first();
$this->I->selectOption($el, $city->id);
````

Here's the output fail file.
``http://pastebin.com/raw.php?i=VNc7Z6Y8``

When using the ``grabTextForm($el)``:
``http://pastebin.com/raw.php?i=M8Xnd1q9``

Thrown InvalidArgumentException from CLI:
``http://pastebin.com/raw.php?i=FT8FeRDq``

This is an acceptance test, this is the config in acceptance.suit.yml:
````
class_name: AcceptanceTester
modules:
    enabled:
        - PhpBrowser
        - AcceptanceHelper
        - Laravel4
        - Db
    config:
        PhpBrowser:
            url: '<I didn't include here, for security purposes>'
        Laravel4:
            cleanup: false
````

Using the input tag works well, while this select tag, really not working, I also tried creating a simple html just to test, but still 2.0.9 was failing to select a value.

Fyi,
I'm using codeception 2.0.9 for my old Laravel4 Framework Project.

**Comment by zbateson** [comment]
As it is nothing's jumping out at me specifically... could you send the simple test you created instead?  That would probably be much easier to see what's going on and if it's a bug, create a test case out of :wink:

**Comment by daison12006013** [comment]
Hi @zbateson, I did blame to your commit in the file src/Codeception/Lib/InnerBrowser.php at line 693, as per checking you're using ``$fieldName = $this->getSubmissionFormFieldName($field->attr('name'));``

While checking function ``getSubmissionFormFieldName()``, it seems it removes the last 2 characters which it is ``[]``

As you can see, most of the fields that i'm using they're inputs with array.

As per checking the ``fillField()`` function, it doesn't call this ``getSubmissionFormFieldName()`` function, that's why ``fillField()`` works well.

I'm currently not in the office right now, it's my day off, so I cant share my simple test. Try to use an array select tag with name such as ``addresses[city_id][]``, I'm sure you will get the same error.

**Comment by zbateson** [comment]
Hi @daison12006013 - no worries... I'm not discounting the possibility that a bug exists but I need clearer steps on recreating it.

Actually removing the last two characters is done not to match the field name in the HTML document but to match the name of the field in the DomCrawler/Form object.  Even ```fillField``` does it, but it doesn't call ```getSubmissionFormField``` it just does it directly (well, calls ```matchFormField``` which you'll notice does exactly the same thing on its first few lines).

**Comment by daison12006013** [comment]
Hi @zbateson, I created a simple tests and pushed to this repo: ``https://github.com/daison12006013/codeception2.0.9-tests``

try to download or clone, and run, I also added their the codecept.phar for v2.0.9.

got to ``tests/acceptains/ExampleCept.php``, i commented out the source code with comments, so try to un-comment and observe the results.

I created 2 select tags, named them as ``city_id`` and ``addresses[city_id][]``

``city_id`` works well, no errors, I was able to select a value. While ``addresses[city_id][]`` fails the same error i've got when using laravel4.

**Comment by Naktibalda** [comment]
Does your issue happen with v2.0.14?

**Comment by daison12006013** [comment]
Hi @Naktibalda,

Yes, v2.0.14 is more worst, it affects the ``fillField()`` for  textarea, you can check my ``SimpleCept.php``, so that you can check the sequence. When commenting the textarea error, I also got a problem with the selectOption.
![screen shot 2015-06-22 at 7 15 30 am](https://cloud.githubusercontent.com/assets/4581415/8273793/a6683d7e-18ae-11e5-8959-57f89413425c.png)

when using ``v2.0.9`` the ``fillField()`` for textarea works well, however the selectOption() is getting error.

any urgent commit fix for this?

**Comment by zbateson** [comment]
Hi @daison12006013 

The issue was with mapping DomCrawler/Form/Field objects, with the following condition:

1. they ended with '[]'
2. you've previously set a value on them (the second fillField/selectOption/checkOption was failing).

Or in general using selectOption on a radio input seems to have been failing.

These should be fixed now in 2.0-dev, and relevant tests added in #2077 .  I will be merging it shortly.
